### DISCUSSION TITLE
confidence level sensitivity

### DISCUSSION THREAD

**Discussion by rover-t** [JamesSG2/Handwriting-Conversion-Application#26] [Ideas]
Hey @JamesSG2 I am wondering if the sensitivity on the confidence level checker may be to high. Getting 50% of erroneous words for 5% of correct words seems good, but the problem I'm noticing is that an erroneous word still has generally 50% correct characters, generally only one or two characters in a given incorrect word is wrong. This means that when looking at the character level about half of the characters deleted are actually still useful.

Since we give users the ability to manually select characters and discard incorrect characters, it may be more effective to provide the user as many options as possible so they can find a suitable choice. We don't need to make any changes, I just wanted this to be something we consider. I think that the result we are getting is still better than without, but I'm wondering if there are further improvements that can be made.

**Comment by JamesSG2** [comment]
Yes, I wondered about this myself earlier which is why I made the CL adjustable in the code. Also I did some tests and got similar result, in which the rejected words are about 50% correct letters and 50% incorrect letters. However this is actually a good thing, since there are usually many more correct letters than incorrect. If you do the math it actually comes out to the sytem rejecting about 60% of all incorrect letters while rejecting only about 20% of correct letters. It's not great that 20% of correct letters are rejected in the process, but my philosophy is that it's much better to be cautious about each new letter the sytem accepts. This is because we want to reduce the correcting the user needs to do as much as possible. In order for this program to be practically usfull it will need to be able to handle large essays, and if 5+ corrections need to be made on each line it won't be very useful. I suggest that for the handwritting input the user should enter the same pangram 10 times on the same paper (they'll only need to do it once). This would make up for any loss of diversity in the letter-images from rejection or other mechanisms, and provide the user / output with many more letter-image options; making the output a bit more random and realistic.
